modules/apps/content-dashboard/content-dashboard-test/src/testIntegration/java/com/liferay/content/dashboard/web/internal/portlet/test/util/MockLiferayPortletRenderRequestUtil.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.content.dashboard.web.internal.portlet.test.util;
+
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.portlet.bridges.mvc.constants.MVCRenderConstants;
+import com.liferay.portal.kernel.security.permission.PermissionThreadLocal;
+import com.liferay.portal.kernel.test.portlet.MockLiferayPortletRenderRequest;
+import com.liferay.portal.kernel.test.portlet.MockLiferayPortletURL;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ProxyUtil;
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.util.Locale;
+import java.util.Objects;
+
+import javax.portlet.PortletContext;
+import javax.portlet.PortletPreferences;
+import javax.portlet.PortletRequestDispatcher;
+
+/**
+ * @author David Arques
+ */
+public class MockLiferayPortletRenderRequestUtil {
+
+	public static MockLiferayPortletRenderRequest
+			getMockLiferayPortletRenderRequest(
+				String[] assetVocabularyNames, Company company, Locale locale)
+		throws Exception {
+
+		MockLiferayPortletRenderRequest mockLiferayPortletRenderRequest =
+			new MockLiferayPortletRenderRequest();
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			WebKeys.COMPANY_ID, company.getCompanyId());
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			StringBundler.concat(
+				mockLiferayPortletRenderRequest.getPortletName(), "-",
+				WebKeys.CURRENT_PORTLET_URL),
+			new MockLiferayPortletURL());
+
+		String path = "/view.jsp";
+
+		mockLiferayPortletRenderRequest.setParameter("mvcPath", path);
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			MVCRenderConstants.
+				PORTLET_CONTEXT_OVERRIDE_REQUEST_ATTIBUTE_NAME_PREFIX + path,
+			ProxyUtil.newProxyInstance(
```

Comment:
Maybe the PortetContext it is worth it to be created as a MockPortletContext in the portal-test lib.

Reply:
Interesting!

Reply:
I'm not sure how to do this, could you explain it a bit more? Maybe it could be handled in another issue, what do you think?

Reply:
If the requirement were to create a MockPortletContext, how would you do it?

Reply:
See https://github.com/liferay-tango/liferay-portal/pull/904